### DISCUSSION TITLE
Is it possible to define a host name so I can just enter it instead of the IP address when adding proxies?

### DISCUSSION THREAD

**Discussion by ghost** [NginxProxyManager/nginx-proxy-manager#4249] [General]
Many of my docker apps are on the same network as NPM so i can just set their docker name as the location. but apps runnign directly on the host I have to use the IP address. is it possible to define a name that translates directly to the ip address, so that I can set the locations of those app to that name, and then if for whatever reason the computer's ip address changes I just have to update one entry, or have that set automatically?

**Comment by leolivier** [comment]
same here but I suppose this can be done in the /data/nginx/custom/root_top.conf by defining a variable there? Didn't check yet

**Comment by nokarukuta** [comment]
Has anyone tried this?

**Comment by Haswell-s** [comment]
If you run as docker you absolutely can!
You have to create a network and make sure all the containers are connected to it, you can verify with `docker network <network-name> inspect`
The name property for each container on the network is your hostname, then do a <hostname>:<port> and it should work.

DO NOTE: your port mapping doesn't matter, you don't even need one since your NPM is talking on the docker-network level. Thus each container is like a separate machine with its own ports -> 2 services can have the same port and it'll look something like:
```
portainer:443
homeassistant:443
```
YOU HAVE TO TALK IN INTERNAL PORTS, exterals aren't recognised inside the docker network.

Man is that clean tho!
